cinema/views.py

```diff
@@ -51,8 +87,63 @@
     def get_serializer_class(self):
         if self.action == "list":
             return MovieSessionListSerializer
-
         if self.action == "retrieve":
             return MovieSessionDetailSerializer
-
         return MovieSessionSerializer
+
+    def get_queryset(self):
+        if self.action == "list":
+            capacity_expr = ExpressionWrapper(
+                F("cinema_hall__rows") * F("cinema_hall__seats_in_row"),
+                output_field=IntegerField())
+            queryset = (self.queryset.prefetch_related("tickets", "movie")
+                        .annotate(
+                tickets_available=capacity_expr - Count("tickets")
+            ))
+            date = self.request.query_params.get("date")
+            movie = self.request.query_params.get("movie")
+            if date:
+                date = parse_date(date)
+                queryset = queryset.filter(show_time__date=date)
+            if movie:
+                movie_ids = [int(str_id) for str_id in movie.split(",")]
+                queryset = queryset.filter(movie_id__in=movie_ids)
+            return queryset.distinct()
+        elif self.action == "retrieve":
+            return self.queryset.prefetch_related(
+                "movie",
+                "cinema_hall",
+                "movie__genres",
+                "movie__actors"
+            )
```

Comment:
Prefetching `movie__genres` and `movie__actors` might be unnecessary if these fields are not directly used in the `retrieve` action. Please verify if these prefetches are needed to optimize database queries.